core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/ColumnPageEncoder.java

```diff
@@ -78,6 +78,14 @@ public DataType getTargetDataType(ColumnPage inputPage) {
     }
   }
 
+  public Encoding getEncodingType() {
+    List<Encoding> currEncodingList = getEncodingList();
+    if (CarbonUtil.isEncodedWithMeta(currEncodingList)) {
+      return currEncodingList.get(0);
+    }
+    return null;
```

Comment:
better not returning null, why not return the list itself?

Reply:
This function will only return value in case of Adaptive Encoding, for any other case it will return null. Also we just want the first element of the list to check the type of encoding.